src/mol-gl/shader/cylinders.vert.ts

```diff
@@ -79,6 +79,18 @@ void main() {
         gl_Position.z = (uProjection * mvPosition).z;
     }
 
+    #if defined(dDualColor) && defined(dRenderVariant_color) && (defined(dColorType_group) || defined(dColorType_groupInstance))
+        // dual-color mixing
+        // - for aColorMode between 0 and 1 use aColorMode to interpolate
+        // - for aColorMode == 2 do nothing, i.e., use vColor
+        // - for aColorMode == 3 use position on cylinder axis to interpolate
+        if (aColorMode <= 1.0){
```

Comment:
What about testing here for special case in which vColor === color2 ? Could this improve performances?

Reply:
maybe, maybe not, it would be very small and hard to measure since generally the bottleneck is in the fragment shader